scripts/tests/sanitycheck/test_testsuite_class.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+# pylint: disable=line-too-long
+# pylint: disable=C0321
+'''
+This test file contains testcases for Testsuite class of sanitycheck
+'''
+import sys
+import os
+
+ZEPHYR_BASE = os.getenv("ZEPHYR_BASE")
+sys.path.insert(0, os.path.join(ZEPHYR_BASE, "scripts/sanity_chk"))
+
+from sanitylib import TestCase
+
+def test_testsuite_add_testcases(class_testsuite):
+    """ Testing add_testcase function of Testsuite class in sanitycheck """
+    # Test 1: Check the list of testcases after calling add testcases function is as expected
+    class_testsuite.SAMPLE_FILENAME = 'test_sample_app.yaml'
+    class_testsuite.TESTCASE_FILENAME = 'test_data.yaml'
+    class_testsuite.add_testcases()
+    expected_testcases = ['scripts/tests/sanitycheck/test_data/testcases/tests/test_b/kernel.fifo.timeout',
+                          'scripts/tests/sanitycheck/test_data/testcases/tests/test_b/kernel.fifo.timeout.poll',
+                          'scripts/tests/sanitycheck/test_data/testcases/tests/test_c/kernel.fifo.usage',
+                          'scripts/tests/sanitycheck/test_data/testcases/tests/test_c/kernel.fifo.usage.poll',
+                          'scripts/tests/sanitycheck/test_data/testcases/tests/test_a/kernel.fifo',
+                          'scripts/tests/sanitycheck/test_data/testcases/tests/test_a/kernel.fifo.poll',
+                          'scripts/tests/sanitycheck/test_data/testcases/samples/test_app/sample.app_dev.external_lib']
+    assert set(class_testsuite.testcases.keys()) == set(expected_testcases)
+
+    # Test 2 : Assert Testcase name is expected & all the testcases values are testcase class objects
+    testcase = class_testsuite.testcases.get('scripts/tests/sanitycheck/test_data/testcases/tests/test_a/kernel.fifo')
+    assert testcase.name == 'scripts/tests/sanitycheck/test_data/testcases/tests/test_a/kernel.fifo'
+    assert all(isinstance(n, TestCase) for n in class_testsuite.testcases.values())
```

Comment:
is there any chance we might not be a testcase object? you have been treating this as such in the previous tests.

Reply:
There is an edge case in add_testcases function where an error could be raised "Cant load the testcase.yaml file", which I wasn't able to reproduce. So just wanted to check that no error is raised  from the try block. Thats why checking for the end result ie.. testcases dictionary keys & values.